generator/src/test/java/com/scottlogic/deg/generator/generation/RegexStringGeneratorTests.java

```diff
@@ -246,6 +248,19 @@ void shouldReturnValuesWhenNonContradictingConstraints() {
         Assert.assertNotEquals(0, nonContradictingGenerator.getValueCount());
     }
 
+    @Test
+    void shouldNotGenerateInvalidUnicodeCodePoints() {
+        StringGenerator generator = new RegexStringGenerator("Test.Test", true);
+        Iterable<String> resultsIterable = generator.generateAllValues();
+        for (String s : resultsIterable) {
+            if (s!=null){
+            for (char c : s.toCharArray()) {
+                if (Character.isSurrogate(c)) {
```

Comment:
Is the for loop necessary? From a glance you are generating the string `TestaTest`, `TestbTest`, `Test@Test` therefore the index of the suspect character will always be `4`. Wouldn't `s.toCharArray()[4]` give you the one character to test?